app/src/main/java/ee/ioc/phon/android/speechutils/AbstractAudioRecorder.java

```diff
@@ -40,25 +45,34 @@
     // 2 (bytes) * 1 (channels) * 30 (max rec time in seconds) * 44100 (times per second) = 2 646 000 bytes
     // but typically is:
     // 2 (bytes) * 1 (channels) * 20 (max rec time in seconds) * 16000 (times per second) = 640 000 bytes
-    private final byte[] mRecording;
+    final byte[] mRecording;
 
     // TODO: use: mRecording.length instead
     private int mRecordedLength = 0;
+    private AtomicLong mRecordedSessionId = new AtomicLong(0L);
+    boolean mRecordingBufferIsFullWithData = false;
```

Comment:
this is a flag that is false while the recording buffer is not full of data and changed to true in the first time that the buffer is full (and the writing to the buffer is cyclic)